src/core/config/tmf.h

```diff
@@ -189,7 +189,12 @@
  * This is compulsory for 1.2 FTD.
  *
  */
+#ifdef OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE
+#error \
+    "Don't define `OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE` manually. It's an alias of `((OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2) && OPENTHREAD_FTD)`."
```

Comment:
```suggestion
    "Don't define `OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE` manually. It's already defined when `((OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2) && OPENTHREAD_FTD)`."
```

Reply:
I would prefer the current wording. 
Thoughts @librasungirl

Reply:
Your wording sounds like `OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE` is not defined if not `((OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2) && OPENTHREAD_FTD)`
But in fact `OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE` is always pre-defined and users should never define it manually.

Reply:
I am fine to keep it.